Cpp2IL.Core/CorePlugin/CallAnalysisProcessingLayer.cs

```diff
@@ -0,0 +1,361 @@
+                                IncreaseCount(unknownCalls, m);
+                            }
+                            else if (list.Count > 1)
+                            {
+                                IncreaseCount(deduplicatedCalls, m);
+                            }
+                            else
+                            {
+                                var calledMethod = list[0];
+                                Add(callsDictionary, m, calledMethod);
+                                Add(calledByDictionary, calledMethod, m);
+                            }
+                        }
+                        else if (!keyFunctionAddresses.IsKeyFunctionAddress(address))
+                        {
+                            IncreaseCount(unknownCalls, m);
+                        }
+                    }
+                    else
+                    {
+                        IncreaseCount(unknownCalls, m);
+                    }
+                }
+            }
+        }
+
+        foreach (var assemblyAnalysisContext in appContext.Assemblies)
+        {
+            var callerCountAttributeInfo = callerCountAttributes[assemblyAnalysisContext];
+            var callsUnknownMethodsAttributeInfo = callsUnknownMethodsAttributes[assemblyAnalysisContext];
+            var callsDeduplicatedMethodsAttributeInfo = callsDeduplicatedMethodsAttributes[assemblyAnalysisContext];
+            var callsAttributeInfo = callsAttributes[assemblyAnalysisContext];
+            var calledByAttributeInfo = calledByAttributes[assemblyAnalysisContext];
+
+            foreach (var m in assemblyAnalysisContext.Types.SelectMany(t => t.Methods))
+            {
+                if (m.CustomAttributes == null || m.Definition == null)
+                    continue;
+
+                AddOneParameterAttribute(m, callerCountAttributeInfo, GetCount(callCounts, m.UnderlyingPointer));
+                var unknownCallCount = GetCount(unknownCalls, m);
+                if (deduplicatedCalls.TryGetValue(m, out var deduplicatedCallCount))
+                {
+                    AddOneParameterAttribute(m, callsDeduplicatedMethodsAttributeInfo, deduplicatedCallCount);
+                }
+                if (callsDictionary.TryGetValue(m, out var callsList))
+                {
+                    foreach (var calledMethod in callsList)
+                    {
+                        var il2cppType = GetTypeFromContext(calledMethod.DeclaringType);
+                        if (il2cppType is not null)
+                        {
+                            AddTwoParameterAttribute(m, callsAttributeInfo, il2cppType, calledMethod.Name);
+                        }
+                        else
+                        {
+                            unknownCallCount++;
+                        }
+                    }
+                }
+                const int MaximumCalledByAttributes = 20;//We don't want 1000 attributes on a single method :)
```

Comment:
At the very least move this to a const field at the top of the file for easy future editing, maybe consider making it configurable at runtime